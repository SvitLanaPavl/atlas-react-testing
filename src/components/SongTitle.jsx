import React from 'react'

const SongTitle = () => {
  return (
    <div>
      <h2 className='mb-2 text-2xl leading-8 font-bold text-customPurple-600'>Painted in Blue</h2>
      <p className='mb-4 text-customPurple-300'>Soul Canvas</p>
    </div>
  )
}

export default SongTitle