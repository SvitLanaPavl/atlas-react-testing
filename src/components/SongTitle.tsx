import React from 'react'

interface SongTitleProps {
  title: string;
  artist: string;
}

const SongTitle: React.FC<SongTitleProps> = ( {title, artist} ) => {
  return (
    <div >
      <h2 className='mb-2 text-2xl leading-8 font-bold text-customPurple-600'>{title}</h2>
      <p className='mb-4 text-customPurple-300'>{artist}</p>
    </div>
  )
}

export default SongTitle