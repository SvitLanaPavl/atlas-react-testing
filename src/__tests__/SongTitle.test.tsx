import { render } from '@testing-library/react';
import { expect, test } from 'vitest';
import SongTitle from '@/components/SongTitle';

test('Song title reders correctly', () => {
  const { container } = render(<SongTitle title='Painted in Blue' artist='Soul Canvas'  />);
  expect(container).toMatchSnapshot();
});