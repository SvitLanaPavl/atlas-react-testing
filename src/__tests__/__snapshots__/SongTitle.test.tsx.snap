// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Song title reders correctly 1`] = `
<div>
  <div>
    <h2
      class="mb-2 text-2xl leading-8 font-bold text-customPurple-600"
    >
      Painted in Blue
    </h2>
    <p
      class="mb-4 text-customPurple-300"
    >
      Soul Canvas
    </p>
  </div>
</div>
`;
